### PR TITLE
Add support for mesa libraries.

### PR DESCRIPTION
**Pull request by kyrofa**
Currently Snapcraft adds only a few standard paths to LD_LIBRARY_PATH. This PR expands that to include paths used by mesa packages.

This will need to be backported to 1.x once approved.

Resolves LP: [#1531620](https://bugs.launchpad.net/snapcraft/+bug/1531620)

### CODE REVIEW

File: snapcraft/tests/test_yaml.py [hunk 1]
```diff
@@ -346,6 +347,107 @@ def test_config_expands_filesets(self, mock_loadPlugin):
...
 
+class TestYamlEnvironment(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        dirs.setup_dirs()
+
+        tempdirObj = tempfile.TemporaryDirectory()
+        self.addCleanup(tempdirObj.cleanup)
+        os.chdir(tempdirObj.name)
```
Comment (elopio) [inline snapcraft/tests/test_yaml.py]: We already have this on the base setup, when this test runs the cwd is already a tempdir that will be deleted. Tha path to that dir is in self.path.
https://github.com/ubuntu-core/snapcraft/blob/master/snapcraft/tests/__init__.py#L33

File: snapcraft/tests/test_yaml.py [hunk 2]
```diff
@@ -346,6 +347,107 @@ def test_config_expands_filesets(self, mock_loadPlugin):
...
+        with open('icon.svg', 'w') as icon:
+            icon.write('<svg />')
```
Comment (elopio) [inline snapcraft/tests/test_yaml.py]: aren't the icons optional now? I can't remember if that landed or not yet.

Comment (kyrofa) [inline snapcraft/tests/test_yaml.py]: Oh right, I forgot about that! I do believe it's landed.

Comment (elopio) [inline snapcraft/tests/test_yaml.py]: ah, yes it is because you don't have it in your example! So this can be removed.

File: snapcraft/tests/test_yaml.py (hunk 1 above)
Comment (kyrofa) [inline snapcraft/tests/test_yaml.py]: Ah good find! I was copying what was already there-- I've fixed the rest of the tests as well.

File: snapcraft/tests/test_yaml.py (hunk 2 above)
Comment (kyrofa) [inline snapcraft/tests/test_yaml.py]: Fixed.

**Comment by elopio** [comment]
lgtm. Maybe, I would move the library handling to a new module. snapcraft/libraries?
I don't like that we have the mesa lib hardcoded and that we'll have to keep adding paths there, but I don't know much about that and can't think of a better way to do it.

Thanks for the example!

**Comment by kyrofa** [comment]
@elopio alright I moved it into `snapcraft.libraries`. I agree with you on the hard-coding, but there doesn't seem to be another way to do it since the .debs typically configure themselves in ld.so.conf.d via postinst scripts, which don't get run with how we're unpacking them.

**Comment by sergiusens** [comment]
I re triggered the examples test run

File: snapcraft/libraries.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015 Canonical Ltd
```
Comment (sergiusens) [inline snapcraft/libraries.py]: 2016

File: snapcraft/tests/test_libraries.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015 Canonical Ltd
```
Comment (sergiusens) [inline snapcraft/tests/test_libraries.py]: 2016

File: snapcraft/tests/test_yaml.py [hunk 5]
```diff
@@ -317,7 +300,6 @@ def test_config_expands_filesets(self, mock_loadPlugin):
 version: "1"
 summary: test
 description: test
-icon: my-icon.png
```
Comment (sergiusens) [inline snapcraft/tests/test_yaml.py]: shouldn't these be a separate commit?

**Comment by sergiusens** [comment]
What wishlist bug is this closing? :wink:

File: snapcraft/tests/test_yaml.py (hunk 5 above)
Comment (kyrofa) [inline snapcraft/tests/test_yaml.py]: They could be. These small refactors were to simply make the tests consistent with the ones I was adding rather than being consistent the other way and adding more cruft.

**Comment by kyrofa** [comment]
@sergiusens heh. Fixed :smiley: .

**Comment by sergiusens** [comment]
:+1:
